pyccel/ast/internals.py

```diff
@@ -182,6 +182,21 @@ class PyccelSymbol(str, Immutable):
     >>> x = PyccelSymbol('x')
     x
     """
+    _is_temp = False
+
+    def __new__(cls, name, is_temp=False):
+        return super().__new__(cls, name)
+
+    def __init__(self, name, is_temp=False):
+        self._is_temp = is_temp
+        super().__init__()
+
+    @property
+    def is_temp(self):
+        """
+        Indicates if the variable created by Pyccel [default value: False]
+        """
+        return self._is_temp
```

Comment:
Please modify this docstring:

- Add `was` after "variable"
- Change "variable" with `symbol`
- Reconsider the meaning of this whole sentence: every symbol is created by Pyccel!
What did you really mean here? Maybe this?
```
Indicates if this symbol represents a temporary variable created by Pyccel,
and was not present in the original Python code.
```

Reply:
yes, exactly.